src/components/ImageView/ContourView/ContourViewComponent.tsx

```diff
@@ -25,14 +25,29 @@ export class ContourViewComponent extends React.Component<ContourViewComponentPr
     componentDidMount() {
         this.contourWebGLService = ContourWebGLService.Instance;
         this.gl = this.contourWebGLService.gl;
+        const contourStream = AppStore.Instance.backendService.contourStream;
         if (this.canvas) {
-            this.updateCanvas();
+            contourStream.subscribe(() => {
+                const animatorStore = AnimatorStore.Instance;
+                const receivedContourStores = Array.from(this.props.frame.contourStores.values());
+                if (receivedContourStores.every(contourImageData => contourImageData.progress === 1) && receivedContourStores.length === this.props.frame.contourConfig.levels.length && animatorStore.serverAnimationActive) {
```

Comment:
Can we simply use `FrameStore.contourProgress`?
Do we consider contours from matched images (`AppStore.contourFrames`)?